students/carlos_novoa/lesson08/circle_class.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import math
+
+"""
+Lesson8 - Circle Class
+"""
+
+
+class Circle(object):
+    version = '0.1'
+
+    def __init__(self, radius=None, diameter=None):
+
+        if radius is None and diameter is None:
+            raise ValueError()
+
+        if radius is not None:
+            try:
+                if radius < 0:
+                    raise ValueError()
+            except ValueError:
+                print('Radius must be positive.')
+                return
+            else:
+                self._radius = radius
+                self._diameter = radius * 2
+                self._area = round(math.pi * radius ** 2.0, 6)
```

Comment:
This calculation should also be moved to the `area` property. That way the area can be recalculated based on the current radius value every time the area attribute is accessed. As this is written, your area will only be defined based on the initial radius value, and will not stay in sync if the radius value is changed